src/Explorer/Web/ContractView.hs

```diff
@@ -55,31 +82,51 @@ extractInfo CStateView cv =
             })
   where res = CJ.resource cv
 
+extractInfo CTxView cv (Just (Transactions txs)) =
+  ContractTxView (CJ.contractId . CJ.resource $ cv) . map convertTx $ txs
+  where
+    convertTx tx = CTVR
+      { ctvrLink = txLink tx
+      , ctvrBlock = txBlock tx
+      , ctvrSlot = txSlot tx
+      , ctvrContractId = txContractId tx
+      , ctvrTransactionId = txTransactionId tx
+      }
+
+extractInfo _ _ Nothing = ContractViewError "Something went wrong, unable to display"
+
+
 allContractViews :: [ContractViews]
-allContractViews = [CInfoView, CStateView]
+allContractViews = [CInfoView, CStateView, CTxView]
 
 getNavTab :: ContractViews -> String
 getNavTab CInfoView = "info"
 getNavTab CStateView = "state"
+getNavTab CTxView = "txs"
 
 getNavTitle :: ContractViews -> String
 getNavTitle CInfoView = "Details"
 getNavTitle CStateView = "Code"
+getNavTitle CTxView = "Transactions"
 
 data ContractViews = CInfoView
                    | CStateView
+                   | CTxView
   deriving (Eq)
 
 data ContractView = ContractInfoView CIVR
                   | ContractStateView CSVR
-                  | ContractViewError String;
+                  | ContractTxView String [CTVR]
```

Comment:
Could we put all this inside the record so that we know what things are and for consistency and maintainability?

Reply:
I don't understand what you're asking for this one. Inside `CTVR`? That list of transactions is sometimes empty so we lose the contract id which is needed to put at the top of the page.

Reply:
This whole area is kind of clunky, not sure what to do with it. The first two tabs are kind of a more uniform type of data. Adding the transactions was different.

Reply:
I would rename the CTVR record to something else, and then add one around with it and the contractId and call it CTVR. For this to be really satisfying to look at, maybe it would be the right thing to extract the contractId out, but while it is only one thing probably not worth it, plus ViewError doesn't have it...

Reply:
Got it. So something like `ContractTxView CTVRs` where `data CTVRs = CTVRs String [CTVR]`
Keeping these "top level" `Contract*View` types cleaner.

Reply:
Exactly :+1: